packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -135,14 +136,18 @@ export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps
           e.preventDefault();
           e.stopPropagation();
           if (focusMode === 'cell' && direction === 'rtl') {
-            focusSafely(ref.current);
+            if (document.activeElement !== ref.current) {
+              focusSafely(ref.current);
+              scrollIntoViewFully(ref.current);
```

Comment:
This pattern of `focusSafely` -> `scrollIntoViewFully` repeats in a couple of other places, musing if I should move the `scrollIntoViewFully` call into `focusSafely`. My concern is that it is possible to be in keyboard modality and have `focusSafely` be triggered NOT by keyboard navigation (e.g. component w/ `autoFocus` renders)